scripts/xtensa-build-zephyr.py

```diff
@@ -0,0 +1,382 @@
+
+	if args.west_path:
+		WEST_TOP=pathlib.Path(args.west_path)
+
+	if args.more_verbose:
+		args.verbose_build = True
+
+	if args.verbose_build:
+		if not args.cmake_args:
+			args.cmake_args = "CMAKE_VERBOSE_MAKEFILE=ON"
+		else:
+			args.cmake_args = "CMAKE_VERBOSE_MAKEFILE=ON " + args.cmake_args
+
+	# split arguments by whitespaces and append -D prefix to match expected CMake format
+	if args.cmake_args:
+		arg_list = shlex.split(args.cmake_args)
+		arg_list = ["-D" + arg for arg in arg_list]
+		args.cmake_args = " ".join(arg_list)
+
+def execute_command(command_args, stdin=None, input=None, stdout=None, stderr=None,
+	capture_output=False, shell=False, cwd=None, timeout=None, check=False, encoding=None,
+	errors=None, text=None, env=None, universal_newlines=None):
+	"""[summary] Provides wrapper for subprocess.run (matches its signature) that prints
+	command executed when 'more verbose' verbosity level is set."""
+	if args.more_verbose:
+		print_args = " ".join(command_args) if isinstance(command_args, list) else command_args
+		print_cwd = f"in: {cwd}" if cwd else f"in: {os.getcwd()}"
+		print(f"Running command: {print_args} {print_cwd}")
+	return subprocess.run(args=command_args, stdin=stdin, input=input, stdout=stdout, stderr=stderr,
+		capture_output=capture_output, shell=shell, cwd=cwd, timeout=timeout,
+		check=check, encoding=encoding, errors=errors, text=text, env=env,
+		universal_newlines=universal_newlines)
+
+def git_submodules_update():
+	global SOF_TOP
+	execute_command("git submodule update --init --recursive", timeout=600, check=True, cwd=SOF_TOP)
+
+def check_west_installation():
+	west_path = shutil.which("west")
+	if not west_path:
+		raise FileNotFoundError("Install west and a west toolchain,"
+			"https://docs.zephyrproject.org/latest/getting_started/index.html")
+	print(f"Found west: {west_path}")
+
+def find_west_workspace():
+	"""[summary] Scans for west workspaces using west topdir command by checking script execution
+	WEST_TOP and SOF_TOP directories.
+
+	:return: [description] Returns path when west workspace had been found, otherwise None
+	:rtype: [type] bool, pathlib.Path
+	"""
+	global WEST_TOP, SOF_TOP
+	paths = [ pathlib.Path(os.getcwd()), SOF_TOP, WEST_TOP]
+	for path in paths:
+		if path.is_dir():
+			result = execute_command("west topdir", stderr=subprocess.DEVNULL,
+				stdout=subprocess.DEVNULL, check=False, cwd=path)
+		if result.returncode == 0:
+			return path
+	return None
+
+def create_zephyr_directory():
+	global WEST_TOP
+	WEST_TOP.mkdir(mode=511, parents=False, exist_ok=False)
+	WEST_TOP = WEST_TOP.resolve(strict=True)
+
+def create_zephyr_sof_symlink():
+	global WEST_TOP, SOF_TOP
+	audio_modules_dir = pathlib.Path(WEST_TOP, "modules", "audio")
+	audio_modules_dir.mkdir(mode=511, parents=True, exist_ok=True)
+	sof_symlink = pathlib.Path(audio_modules_dir, "sof")
+	# Symlinks creation requires administrative privileges in Windows or special user rights
```

Comment:
Or you could just drop the symlink feature on Windows? It should be optional, the older shell version of this was designed to support a variety of setups, users were allowed to organize their directories as they wished.

I'm worried symlinks on Windows will confuse users and generate "support requests", what do you think?

Reply:
Yes of course symlinks are not something commonly seen on Windows. It even surprised me that there is something like this. Running python with administrator rights is a pain and adding user a security policy violation is not a great approach either. I just did not find any better alternative to this. Simple "shortcut" in Windows terms won't do the job and we need to have sof as module in zephyr. So...
Well, we could clone zephyrproject using west "outside of" cloned sof repo but this seems dangerous too.
If I just assume that I can clone zephyr to your SOF_REPO/../../../../ this would be very bold assumption, we can't do that.
I just can't think of any better and easy solution for now and we need that script to start working on Windows.
Maybe lets leave it for now as is? Or maybe you have some alternatives in mind?

Reply:
You know Windows, I don't know it much. We can start with this and see what other people think as long as users can clone in any way they want and use this script only for building and not for cloning.